androidApp/src/main/java/com/mbta/tid/mbta_app/android/component/ErrorBanner.kt

```diff
@@ -0,0 +1,168 @@
+package com.mbta.tid.mbta_app.android.component
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.rounded.Clear
+import androidx.compose.material.icons.rounded.Refresh
+import androidx.compose.material3.Button
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.mbta.tid.mbta_app.android.R
+import com.mbta.tid.mbta_app.model.ErrorBannerState
+import com.mbta.tid.mbta_app.repositories.MockErrorBannerStateRepository
+import com.mbta.tid.mbta_app.repositories.MockSettingsRepository
+import kotlin.time.Duration.Companion.minutes
+import kotlinx.datetime.Clock
+
+@Composable
+fun ErrorBanner(vm: ErrorBannerViewModel) {
+    val state by vm.errorState.collectAsState()
+    when (state) {
+        is ErrorBannerState.DataError -> {
+            ErrorCard(
+                details = { Text(stringResource(R.string.error_loading_data)) },
+                button = {
+                    RefreshButton(label = stringResource(R.string.reload_data)) {
+                        (state as ErrorBannerState.DataError).action()
+                        vm.clearState()
+                    }
+                }
+            )
+        }
+        is ErrorBannerState.NetworkError -> {
+            ErrorCard(
+                details = {
+                    Row(verticalAlignment = Alignment.CenterVertically) {
+                        Icon(
+                            Icons.Rounded.Clear,
+                            contentDescription =
+                                "Displayed when the phone is not connected to the network"
```

Comment:
suggestion: Add the contentDescription to strings.xml so that it is localized. It looks like this is pulling from the comment field of the iOS implementation, but that is a commenter for translators to have context, whereas contentDescription is what should be read aloud to users using TalkBack. 

It looks like there is no description set for this image on the iOS side. I think it would be OK to omit here since it is a decorative element redundant to the "Unable to connect" text, but you can also check with design.